pointer/pointer_test.go

```diff
@@ -63,9 +62,230 @@ func TestAllPtrFieldsNil(t *testing.T) {
 	for i, tc := range testCases {
 		name := fmt.Sprintf("case[%d]", i)
 		t.Run(name, func(t *testing.T) {
-			actualErr := AllPtrFieldsNil(tc.obj)
-			if !reflect.DeepEqual(tc.expected, actualErr) {
-				t.Errorf("%s: expected %t, got %t", name, tc.expected, !tc.expected)
+			if actual := AllPtrFieldsNil(tc.obj); actual != tc.expected {
+				t.Errorf("%s: expected %t, got %t", name, tc.expected, actual)
+			}
+		})
+	}
+}
+
+func TestInt32Ptr(t *testing.T) {
```

Comment:
I feel like these tests are harder to read because of the table-driven testing, while there are only two tests in each. Would it be better to just unwind the loop?

Reply:
Thanks for reviewing. Yes that makes sense. I've updated the tests.

Reply:
That's a lot of temporary variables, but that's fine :-)

/lgtm
/approve